src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy

```diff
@@ -1941,4 +2309,71 @@ class ActionDelegate {
         }
     }
 
+    String makeUrl(String publicViewUrl = publicViewProperties.url, String identifier) {
+        return "${publicViewUrl}/${Base64.getEncoder().encodeToString(identifier.bytes)}" as String
+    }
+
+    void updateMultichoiceWithCurrentNode(MultichoiceMapField field, UriNode node) {
+        List<String> splitPathList = splitUriPath(node.uriPath)
+
+        change field value { splitPathList }
+        change field options { findOptionsBasedOnSelectedNode(node, splitPathList) }
```

Comment:
Maybe it would be better to change the order? If exception if thrown in `findOptionsBasedOnSelectedNode` then you would end up with multichoice field with value but no options.